cloudpickle/cloudpickle.py

```diff
@@ -800,6 +787,12 @@ def save_global(self, obj, name=None, pack=struct.pack):
         elif obj in _BUILTIN_TYPE_NAMES:
             return self.save_reduce(
                 _builtin_type, (_BUILTIN_TYPE_NAMES[obj],), obj=obj)
+
+        decomposed_generic = _try_decompose_generic(obj)
```

Comment:
We are doing a whole lot of type checking in `_try_decompose_generic`/`_try_decompose_special_generic`. How about splitting those functions apart and register unitary reducers in the `dispatch_table` instead?

Reply:
I think I can do that, will try.

Reply:
Does [this](/cloudpipe/cloudpickle/pull/318/commits/bc0c368b7e0309a6b6b1da9b6c41b9b4408bf78c) work? I'm not happy with `typing._trim_name(typing._qualname(type(obj)))` but it's the only way I could figure it out. The root of special forms implement `__reduce__`, but I don't think it can be used from inside the dispatch table?